git.go

```diff
@@ -90,67 +90,92 @@ func (r gitReferencer) References() ([]*model.Reference, error) {
 	})
 }
 
-// maybeRoot is a commit hash that may or may not be a root commit.
-type maybeRoot struct {
-	hash   model.SHA1
-	isRoot bool
+type commitFrame struct {
+	cursor int
+	hashes []plumbing.Hash
+	roots  [][]model.SHA1
 }
 
-func rootCommits(r *git.Repository, from plumbing.Hash, seen map[model.SHA1][]model.SHA1) ([]model.SHA1, error) {
-	commit, err := r.CommitObject(from)
-	if err != nil {
-		return nil, err
+func rootCommits(
```

Comment:
As soon as implementation is not trivial, do you think adding a comment here i.e with the link to an awesome explanation at https://github.com/src-d/borges/pull/165 would make sense?

It might simplify life for the future maintainers by avoiding `git blame` (which might or might not immediately lead to the PR).

Reply:
Yes, I think it makes a lot of sense to put it there. I'll add some sort of docs with a brief explanation of what it does :+1: